sdk/eventhub/Azure.Messaging.EventHubs/tests/Diagnostics/DiagnosticsLiveTests.cs

```diff
@@ -0,0 +1,555 @@
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="TrackOne.EventHubsDiagnosticSource" /> fires
+        ///   events as expected.
+        /// </summary>
+        ///
+        [Test]
+        public async Task SendDoesNotInjectContextWhenNoListeners()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    var eventQueue = CreateEventQueue();
+
+                    using (var listener = CreateEventListener(eventQueue))
+                    using (var subscription = SubscribeToEvents(listener))
+                    {
+                        var parentActivity = new Activity("RandomName").AddBaggage("k1", "v1").AddBaggage("k2", "v2");
+                        var eventData = new EventData(Encoding.UTF8.GetBytes("I hope it works!"));
+
+                        // Disable all events.
+
+                        listener.Disable();
+
+                        // Assert that the properties we will check are not already included.
+
+                        Assert.That(eventData.Properties.ContainsKey(TrackOne.EventHubsDiagnosticSource.ActivityIdPropertyName), Is.False);
+                        Assert.That(eventData.Properties.ContainsKey(TrackOne.EventHubsDiagnosticSource.CorrelationContextPropertyName), Is.False);
+
+                        // Send the event.
+
+                        parentActivity.Start();
+
+                        await producer.SendAsync(eventData);
+
+                        parentActivity.Stop();
+
+                        // There should be no Diagnostic-Id injection.
+
+                        Assert.That(eventData.Properties.ContainsKey(TrackOne.EventHubsDiagnosticSource.ActivityIdPropertyName), Is.False);
+
+                        // There should be no Correlation-Context injection.
+
+                        Assert.That(eventData.Properties.ContainsKey(TrackOne.EventHubsDiagnosticSource.CorrelationContextPropertyName), Is.False);
+
+                        // There should be no more events to dequeue.
+
+                        Assert.That(eventQueue.TryDequeue(out var evnt), Is.False);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="TrackOne.EventHubsDiagnosticSource" /> fires
+        ///   events as expected.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null, false)]
+        [TestCase(null, true)]
+        [TestCase("AmIaGoodPartitionKey", false)]
+        public async Task SendFiresExceptionEvents(string partitionKey, bool usePartitionId)
```

Comment:
extreme nit:  Would you mind breaking the parameters onto separate lines for consistency with the approach used elsewhere?